lesson/src/App.js

```diff
@@ -0,0 +1,104 @@
+        { id: 2, isRented: false, title: "Beauty and the Beast", year: 1991, img: "https://images-na.ssl-images-amazon.com/images/I/81etFyb9N-L._SL1500_.jpg", descrShort: "A kickass woman named Belle who does not succumb to social norms gets crap from a bunch of village idiots, chief amongst them a total tool named Gaston. Belle shows everyone how great she is when she turns a beast (not Gaston) into a man. Love ensues, but then the villagers fall trap to severe group-think mentality led by the main tool himself." },
+        { id: 3, isRented: false, title: "The Sword in the Stone", year: 1963, img: "https://www.disneyinfo.nl/images/laserdiscs/229-1-AS-front.jpg", descrShort: "Arthur is a young boy who just wants to be a knight's squire. Alas, he is dubbed 'Wart' early on, and it was all downhill from there for a while. On a hunting trip he falls in on Merlin, literally. Merlin is a possibly-mentally-unstable-and-ethically-dubious Wizard that turns Arthur into a literate, at-one-point harassed squirrel. Watch to find out what the heck that means." },
+        { id: 4, isRented: false, title: "Beauty and the Beast", year: 2016, img: "https://images-na.ssl-images-amazon.com/images/I/51ArFYSFGJL.jpg", descrShort: "Basically the same as the original, except now Hermi-- Emma Wattson plays Belle, fittingly so some would say, given how actively progressive she is regarding women's rights. Rumor has it that in the bonus scenes she whips out a wand and turns Gaston into a toad, but in order to watch those scenes you need to recite a certain incantation." }
+      ],
+      users:[
+        {name:"Aubida Naalwa",color:"red", pudget:2000},
+        {name :"Alaa Naalwa",color:"blue",pudget:2000},
+        {name:"Noor Masarwe",color:"green",pudget:2000},
+        {name:"Amro Habib",color:"white",pudget:2000}
+      ],
+      loggedIn:false,
+      currentUser:0
+    }
+
+
+  }
+
+  setUserLoggedIn = (userName) => {
+    const index = this.state.users.findIndex(v => v.name == userName)
+    this.setState({
+      loggedIn:true,
+      currentUser:index
+    })
+  }
+
+  removeFromRent =(user, movie)=>{
+    if(!user || !user.name || movie == undefined){
+      return
+    }
+    let users = [...this.state.users]
+    const userFound ={...this.state.users.find(v => user.name == v.name )}
+    if(!userFound || userFound["movies"] == undefined){
+      return
+    }
+    
+  userFound["movies"].splice(userFound["movies"].findIndex(v => v.id == movie.id), 1)
+  users[userFound.id] = userFound
+  this.setState({
+    users
+  })
+   
+  }
+
+
+  addMovieToUser=(user, movie)=>{
+    if(!user || !user.name  || movie == undefined){
```

Comment:
you have this line of code a few times. If you do this over and over it might be better to create a helper function called `checkUserName()` or somthing